File: javascript/1614.Maximum-Nesting-Depth-of-the-Parentheses.js
/**
 * @param {string} s
 * @return {number}
 */
var maxDepth = function (s) {
  let maxCount = 0;
  let currentCount = 0;

  for (const c of s) {
    if (c === "(") {
      currentCount += 1;
    } else if (c === ")") {
      currentCount -= 1;
    }

    maxCount = Math.max(maxCount, currentCount);
  }

  return maxCount;
};

File: python/1614.Maximum-Nesting-Depth-of-the-Parentheses.py
class Solution:
    def maxDepth(self, s: str) -> int:
        max_count = 0
        cur_count = 0

        for c in s:
            if c == "(":
                cur_count += 1
            elif c == ")":
                cur_count -= 1
            max_count = max(max_count, cur_count)
        
        return max_count